### DISCUSSION TITLE
How to `AddWorkflows` from different assemblies?

### DISCUSSION THREAD

**Discussion by wakuflair** [elsa-workflows/elsa-core#5205] [Q&A]
In a modular application, each project(assembly) may have its own workflows to register. But it seems elsa can only register workflows in a single place: `services.AddElsa(cfg => cfg.AddWorkflowsFrom<CurrentClass>)`.

So how can I add workflows in the different assemblies?

**Comment by glime-ai** [comment]
To register workflows from different assemblies in Elsa, you can use the `AddWorkflowsFromAssembly` extension method. This method allows you to specify the assembly containing the workflows you want to register. You can call this method multiple times, once for each assembly that contains workflows you wish to include.

Here's an example of how you can register workflows from multiple assemblies:

```csharp
services.AddElsa(elsa => {
    // Register workflows from the current assembly
    elsa.AddWorkflowsFrom<CurrentClass>();

    // Register workflows from another assembly
    var anotherAssembly = typeof(AnotherClass).Assembly;
    elsa.AddWorkflowsFromAssembly(anotherAssembly);

    // Register workflows from yet another assembly
    var yetAnotherAssembly = typeof(YetAnotherClass).Assembly;
    elsa.AddWorkflowsFromAssembly(yetAnotherAssembly);

    // ... additional configuration
});
```

In the example above, `AnotherClass` and `YetAnotherClass` are classes that reside in the assemblies from which you want to register workflows. By calling `typeof(AnotherClass).Assembly`, you get a reference to the assembly containing `AnotherClass`, and you can do the same for any other class in other assemblies.

This way, you can modularly add workflows from different projects within your solution.

**Comment by wakuflair** [comment]
Thank you AI, but I can't put all assemblies in one place. My assemblies are kind of plugins and loaded dynamically.
